app/components/primary_nav_component.rb

```diff
@@ -8,12 +8,13 @@ class PrimaryNavComponent < ViewComponent::Base
   class NavItemComponent < ViewComponent::Base
     attr_reader :path
 
-    def initialize(path:)
+    def initialize(path:, selected: false)
       @path = path
+      @selected = selected
     end
 
     def current_section?(path)
-      request.path.include?(path)
+      @selected || request.path.include?(path)
```

Comment:
That looks a bit scary. I think there should only ever be one `current_section` element - but with this change, there could be many. Well, technically there could be many beforehand too, if you got very unlucky with paths.

I wonder if we should change the logic so that `path` is only used for navigation, and `selected` is the only thing responsible for selecting current section. Then you could still have multiple current sections, but at least it would be pretty obvious from reading template code that it's happening.

Reply:
Yes, I think this would need some more thought as it looks like this is normally used as content for a layout where it's not known which item is selected? Feels like refactoring that shouldn't be in this PR?

Reply:
Good idea, Tech-debt ticket created